child_compassion/model/child_compassion.py

```diff
@@ -340,12 +374,19 @@ def _get_case_study(self, cr, uid, child, context=None):
             context = dict()
         url = self.get_url(child.code, 'casestudy')
         r = requests.get(url)
-        json_data = r.json()
-        if not r.status_code / 100 == 2:
+        html_res = r.text
+        json_data = dict()
+        error = r.status_code != 200
+        try:
+            json_data = r.json()
+        except:
+            error = True
+        if error:

```

Comment:
You could raise the custom exception from inside the except block. What is the reason to use a flag rather than directly do the job ?